test/sd-test-srv/main.go

```diff
@@ -22,11 +40,15 @@ func dnsHandler(w dns.ResponseWriter, r *dns.Msg) {
 		w.WriteMsg(m)
 		return
 	}
-	if !strings.HasSuffix(r.Question[0].Name, ".boulder.") {
+	if r.Question[0].Qtype != dns.TypeA {
```

Comment:
Is there a reason to avoid proxying non-A record look-ups through the proxy? Right now we're only using A lookups in our containers for external services but I can imagine a day in which we'd need to add `&& != dns.TypeAAAA` and maybe its better to be future proof and invert this logic so that all lookups we know we can't answer ourselves are proxied by default and all A lookups that don't end in `.boulder.` are proxied?

Reply:
Good point; I initially thought it would be complicated to proxy multiple query types, but it's pretty straightforward. I'll make that change.